src/results-output/results-doc.ts

```diff
@@ -236,85 +236,143 @@ export function appendCurrentTopLevelForm() {
     appendFormGrabbingSessionAndNS(true);
 }
 
+function lastLineIsEmpty(doc: vscode.TextDocument): boolean {
+  const lines = doc.lineCount;
+  const lastLine = doc.getText(new vscode.Range(lines - 1, 0, lines - 1, Infinity));
+  return lastLine === "";
+}
+
 let scrollToBottomSub: vscode.Disposable;
 export interface OnAppendedCallback {
     (insertLocation: vscode.Location, newPosition?: vscode.Location): any
 }
-let editQueue: [string, OnAppendedCallback][] = [];
-let applyingEdit = false;
-/* Because this function can be called several times asynchronously by the handling of incoming nrepl messages and those,
-   we should never await it, because that await could possibly not return until way later, after edits that came in from elsewhere
-   are also applied, causing it to wait for several edits after the one awaited. This is due to the recursion and edit queue, which help
-   apply edits one after another without issues.
-
-   If something must be done after a particular edit, use the onResultAppended callback. */
+
+type EditQueueItem = {
+  line: string ;
+  onAppended?: OnAppendedCallback;
+};
+type QueuedCallback = (item: EditQueueItem) => void;
+export class EditQueue {
+  #queue: EditQueueItem[] = [];
```

Comment:
Can this be done more functionally instead? I like to avoid classes, especially ones that mutate internal state, as they seem to add complexity with no real benefit (at least in cases like this). If you disagree, I'd be happy to hear you out and discuss it further.